lib/parser/lexer.rl

```diff
@@ -105,6 +105,8 @@ class Parser::Lexer
     @tokens     = nil
     @comments   = nil
 
+    @has_encode = ''.respond_to?(:encode)
```

Comment:
I think this would be clearer as: `String.instance_methods.include?(:encode)`. Its semantically the same, but does avoid the otherwise unusued literal. Performance wise not relevant suggestion.